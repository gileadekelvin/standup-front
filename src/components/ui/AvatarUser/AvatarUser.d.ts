export type AvatarUserProps = {
  name: string;
};
