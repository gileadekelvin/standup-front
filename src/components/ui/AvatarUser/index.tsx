import { Avatar, Typography } from '@mui/joy';

import { AvatarUserProps } from './AvatarUser';

const AvatarUser = (props: AvatarUserProps) => {
  const { name } = props;
  return (
    <>
      <Avatar size='sm'>
        <Typography fontSize='sm' fontWeight='lg'>
          {name.slice(0, 2).toUpperCase()}
        </Typography>
      </Avatar>
      <Typography fontSize='sm' fontWeight='lg'>
        {name}
      </Typography>
    </>
  );
};

export default AvatarUser;
