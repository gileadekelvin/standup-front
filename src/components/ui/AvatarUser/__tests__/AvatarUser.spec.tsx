import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import AvatarUser from '..';

describe('Test AvatarUser component', () => {
  it('should render component', () => {
    const { container } = render(<AvatarUser name='John' />);

    expect(container).toMatchSnapshot();
  });
});
