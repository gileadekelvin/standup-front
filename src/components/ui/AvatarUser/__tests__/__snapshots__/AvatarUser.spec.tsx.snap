// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AvatarUser component should render component 1`] = `
<div>
  <div
    class="JoyAvatar-root JoyAvatar-variantSoft JoyAvatar-colorNeutral JoyAvatar-sizeSm css-3fr2zd-JoyAvatar-root"
  >
    <p
      class="JoyTypography-root JoyTypography-body1 css-1837725-JoyTypography-root"
    >
      JO
    </p>
  </div>
  <div
    class="JoyBox-root css-j7qwjs"
  >
    <p
      class="JoyTypography-root JoyTypography-body1 css-yfw4hu-JoyTypography-root"
    >
      John
    </p>
    <span
      class="JoyTypography-root JoyTypography-body3 css-3yimt6-JoyTypography-root"
    />
  </div>
</div>
`;
